ai/mcp/src/main/java/com/linecorp/armeria/server/ai/mcp/ArmeriaStreamableServerTransportProvider.java

```diff
@@ -0,0 +1,484 @@
+                              .contextWrite(writeContext(mcpContext))
+                              .toFuture()
+                              .thenApply(unused -> HttpResponse.of(ResponseHeaders.of(HttpStatus.ACCEPTED)));
+            } else if (message instanceof McpSchema.JSONRPCNotification notification) {
+                return session.accept(notification)
+                              .contextWrite(writeContext(mcpContext))
+                              .toFuture()
+                              .thenApply(unused -> HttpResponse.of(ResponseHeaders.of(HttpStatus.ACCEPTED)));
+            } else if (message instanceof McpSchema.JSONRPCRequest rpcRequest) {
+                final Flux<ServerSentEvent> events = Flux.<ServerSentEvent>create(sink -> {
+                    final ArmeriaStreamableMcpSessionTransport transport =
+                            new ArmeriaStreamableMcpSessionTransport(ctx, sink);
+                    final Mono<Void> stream = session.responseStream(rpcRequest, transport);
+                    final Disposable streamSubscription =
+                            stream.onErrorComplete(err -> {
+                                      sink.error(err);
+                                      return true;
+                                  }).contextWrite(sink.contextView())
+                                  .subscribe();
+                    sink.onCancel(streamSubscription);
+                }).contextWrite(writeContext(mcpContext));
+                final HttpResponse response = ServerSentEvents.fromPublisher(events);
+                return UnmodifiableFuture.completedFuture(response);
+            } else {
+                final HttpResponse response =
+                        HttpResponse.ofJson(HttpStatus.BAD_REQUEST, JsonRpcResponse.ofFailure(
+                                JsonRpcError.INVALID_REQUEST.withData("Unknown message type")));
+                return UnmodifiableFuture.completedFuture(response);
+            }
+        }
+
+        @Override
+        protected HttpResponse doDelete(ServiceRequestContext ctx, HttpRequest req) throws Exception {
+            if (isClosing) {
+                return HttpResponse.of(HttpStatus.SERVICE_UNAVAILABLE, MediaType.PLAIN_TEXT,
+                                       "Server is shutting down");
+            }
+
+            final String sessionId = req.headers().get(HttpHeaders.MCP_SESSION_ID, "");
+            if (sessionId.isEmpty()) {
+                return HttpResponse.of(HttpStatus.BAD_REQUEST, MediaType.PLAIN_TEXT,
+                                       "Missing required header: " + HttpHeaders.MCP_SESSION_ID);
+            }
+
+            if (disallowDelete) {
+                return HttpResponse.of(HttpStatus.METHOD_NOT_ALLOWED, MediaType.PLAIN_TEXT,
+                                       "Deleting sessions is not allowed");
+            }
+
+            final McpStreamableServerSession session = sessions.get(sessionId);
```

Comment:
Question) Is there no need to delete `session` from sessions here? I'm assuming the `TransportProvider` lifecycle is the same as the `Server` (or potentially longer)

Reply:
Nice catch, thanks. It should be removed but it seems to have been omitted accidentally while refactoring the upstream code.